onnxruntime/core/optimizer/computation_reduction.cc

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/safeint.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/utils.h"
+#include "core/optimizer/computation_reduction.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+typedef std::function<Status(Graph&, Node&, Node&)> Handler;
+
+static const int GATHERND_BATCH_DIM = 1;
```

Comment:
>static [](start = 0, length = 6)

constexpr is always complier-time constant but static const can be runtime value. Let's use constexpr here.